File: 231-power-of-two/power-of-two.cpp
class Solution {
public:
    bool is_pow(int n){
        if(n<=0){
            return false;
        }
        if(n==1){
            return true;
        }

        if(n%2 == 0){
            return is_pow(n/2);
        }

        return false;
    }

    bool isPowerOfTwo(int n) {

        return is_pow(n);
    }
};

File: 231-power-of-two/power-of-two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        def is_pow(num):
            if num <= 0:
                return False
            
            if num == 1:
                return True
            
            if num%2 == 0:
                return is_pow(num//2)
            
            return False

        return is_pow(n)